typescript/sdk/src/core/EvmCoreModule.ts

```diff
@@ -26,42 +32,92 @@ import { EvmCoreReader } from './EvmCoreReader.js';
 import { EvmIcaModule } from './EvmIcaModule.js';
 import { HyperlaneCoreDeployer } from './HyperlaneCoreDeployer.js';
 import { CoreFactories } from './contracts.js';
+import { CoreConfigSchema } from './schemas.js';
 
-export type DeployedCoreAdresses = HyperlaneAddresses<CoreFactories> & {
-  testRecipient: Address;
-  timelockController?: Address; // Can be optional because it is only deployed if config.upgrade = true
-  interchainAccountRouter: Address;
-  interchainAccountIsm: Address;
-} & HyperlaneAddresses<ProxyFactoryFactories>;
+type DeployedCoreAddresses = Partial<
+  HyperlaneAddresses<CoreFactories> & {
+    testRecipient: Address;
+    timelockController: Address;
+    interchainAccountRouter: Address;
+    interchainAccountIsm: Address;
+  } & HyperlaneAddresses<ProxyFactoryFactories>
+>;
 
 export class EvmCoreModule extends HyperlaneModule<
   ProtocolType.Ethereum,
   CoreConfig,
-  DeployedCoreAdresses
+  DeployedCoreAddresses
 > {
   protected logger = rootLogger.child({ module: 'EvmCoreModule' });
   protected coreReader: EvmCoreReader;
   public readonly chainName: string;
 
-  protected constructor(
+  // We use domainId here because MultiProvider.getDomainId() will always
+  // return a number, and EVM the domainId and chainId are the same.
+  public readonly domainId: Domain;
+
+  constructor(
     protected readonly multiProvider: MultiProvider,
-    args: HyperlaneModuleParams<CoreConfig, DeployedCoreAdresses>,
+    args: HyperlaneModuleParams<CoreConfig, DeployedCoreAddresses>,
   ) {
     super(args);
     this.coreReader = new EvmCoreReader(multiProvider, this.args.chain);
     this.chainName = this.multiProvider.getChainName(this.args.chain);
+    this.domainId = multiProvider.getDomainId(args.chain);
   }
 
   /**
    * Reads the core configuration from the mailbox address specified in the SDK arguments.
    * @returns The core config.
    */
   public async read(): Promise<CoreConfig> {
+    assert(this.args.addresses.mailbox, 'Mailbox not provided for read');
     return this.coreReader.deriveCoreConfig(this.args.addresses.mailbox);
   }
 
-  public async update(_config: CoreConfig): Promise<AnnotatedEV5Transaction[]> {
-    throw new Error('Method not implemented.');
+  /**
+   * Updates the core contracts with the provided configuration.
+   *
+   * @param expectedConfig - The configuration for the core contracts to be updated.
+   * @returns An array of Ethereum transactions that were executed to update the contract.
+   */
+  public async update(
```

Comment:
`update` feels like a misnomer here– it's not actually making the update and therefore I would not expect an "update" function to return _unexecuted_ transactions

Reply:
maybe just "update" does the submission, but we need a function that is explicit to getting these txs (and not executing them)

Reply:
it's not sexy but maybe just `getUpdateTxs`

Reply:
I sorta agree. 

it can update _and/or_ deploy. In the case of some Isms, it can deploy new ones, and then return some update tx (such as setDefaultIsm). I'm ok keeping it `update()` for now

Reply:
btw, `update()` is what we decided when we designed the "crud" pattern. It's also part of the abstract definition. We can definitely change this later

Reply:
Can you create a sim to track?

Reply:
https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/4222